internal/reminder/reminder.go

```diff
@@ -74,21 +78,52 @@ func (r *reminder) Start(ctx context.Context) error {
 		return errors.New("reminder stopped, cannot start again")
 	default:
 	}
+	defer r.Stop()
 
 	interval := r.cfg.RecurrenceConfig.Interval
 	if interval <= 0 {
 		return fmt.Errorf("invalid interval: %s", r.cfg.RecurrenceConfig.Interval)
 	}
 
+	metricsServerDone := make(chan struct{})
+
+	if r.cfg.MetricsConfig.Enabled {
+		metricsProviderReady := make(chan struct{})
+
+		go func() {
+			if err := r.startMetricServer(ctx, metricsProviderReady); err != nil {
+				logger.Fatal().Err(err).Msg("failed to start metrics server")
+			}
+			close(metricsServerDone)
+		}()
+
+		select {
+		case <-metricsProviderReady:
+			var err error
+			r.metrics, err = metrics.NewMetrics(otel.Meter("reminder"))
+			if err != nil {
+				return err
+			}
+		case <-ctx.Done():
+			logger.Info().Msg("reminder stopped")
+			return nil
+		}
```

Comment:
I hope this isn't me too complicating stuff.

Reply:
I do think you're being overly conscientious for shutdown from `main`, but I'll allow it.

Reply:
Thank you, kindly!